marimo/_ast/visitor.py

```diff
@@ -75,6 +79,7 @@ class RefData:
 class ScopedVisitor(ast.NodeVisitor):
     def __init__(self, mangle_prefix: Optional[str] = None) -> None:
         self.block_stack: list[Block] = [Block()]
+        self.ref_stack: list[set[Name]] = [set()]
```

Comment:
Can you add a comment describing the semantics of `ref_stack`?

Reply:
Added a comment here, but even more detail under variabledata.required_refs